packages/mobile/src/components/core/TextButton.tsx

```diff
@@ -16,34 +16,37 @@ import { makeStyles } from 'app/styles'
 
 import type { TextProps } from './Text'
 
-const useStyles = makeStyles(({ palette, spacing }, { variant }) => {
-  const variantStyles = {
-    primary: {
-      icon: {
-        fill: palette.primary
-      }
-    },
-    secondary: {
-      icon: {
-        fill: palette.secondary
-      }
-    },
-    neutralLight4: {
-      icon: {
-        fill: palette.neutralLight4
+const useStyles = makeStyles<Pick<TextButtonProps, 'variant'>>(
+  // @ts-ignore nested NamedStyles<NamedStyles<T>> issue
```

Comment:
sorry what's up here?

Reply:
```
Argument of type '({ palette, spacing }: Theme, { variant }: Pick<TextButtonProps, "variant">) => { root: { flexDirection: string; alignItems: string; }; iconLeft: { marginRight: number; }; iconRight: { ...; }; disabled: { ...; }; } & ({ ...; } | ... 1 more ... | { ...; })' is not assignable to parameter of type 'Styles<NamedStyles<any>, Pick<TextButtonProps, "variant">>'.
  Type '{ root: { flexDirection: string; alignItems: string; }; iconLeft: { marginRight: number; }; iconRight: { marginLeft: number; }; disabled: { color: string; }; } & ({ icon: { fill: string; }; } | { icon: { fill: string; }; } | { ...; })' is not assignable to type 'NamedStyles<any> | NamedStyles<NamedStyles<any>>'.
    Type '{ root: { flexDirection: string; alignItems: string; }; iconLeft: { marginRight: number; }; iconRight: { marginLeft: number; }; disabled: { color: string; }; } & { icon: { fill: string; }; }' is not assignable to type 'NamedStyles<any> | NamedStyles<NamedStyles<any>>'.
      Type '{ root: { flexDirection: string; alignItems: string; }; iconLeft: { marginRight: number; }; iconRight: { marginLeft: number; }; disabled: { color: string; }; } & { icon: { fill: string; }; }' is not assignable to type 'NamedStyles<NamedStyles<any>>'.
        Property 'root' is incompatible with index signature.
          Type '{ flexDirection: string; alignItems: string; }' is not assignable to type 'ViewStyle | TextStyle | ImageStyle'.
            Type '{ flexDirection: string; alignItems: string; }' is not assignable to type 'ImageStyle'.
              Types of property 'alignItems' are incompatible.
                Type 'string' is not assignable to type 'FlexAlignType | undefined'.
```

Reply:
I couldn't figure this one out last night. If you want to take a crack at it or huddle, I'm happy to try again

Reply:
Ah put a "as const" after the textAlign value

Reply:
Oh, nice. That works for those but uncovered another issue which I also saw in `Button.tsx`. I think the issue is that `type NamedStyles<T> = { [P in keyof T]: ViewStyle | TextStyle | ImageStyle }` doesn't properly account for styles meant for icons.

Reply:
gotchaa, yeah makes sense, well whatever is easiest for now :)